test/font_test.py

```diff
@@ -113,6 +113,68 @@ def test_match_font_all_exist(self):
             self.assertFalse(path is None)
             self.assertTrue(os.path.isabs(path))
 
+    def test_match_font_name(self):
+        """that match_font accepts names of various types"""
+        font = pygame_font.get_fonts()[0]
+        not_a_font = "thisisnotafont"
+
+        # Check for not found.
+        self.assertTrue(pygame_font.match_font(not_a_font) is None)
+
+        # Check single name string:
+        font_path = pygame_font.match_font(font)
+        self.assertFalse(font_path is None)
+        self.assertTrue(pygame_font.match_font(not_a_font) is None)
+
+        # Check string of comma-separated names.
+        names = ",".join([not_a_font, font, not_a_font])
+        font_path_2 = pygame_font.match_font(names)
+        self.assertEqual(font_path_2, font_path)
+        names = ",".join([not_a_font, not_a_font, not_a_font])
+        self.assertTrue(pygame_font.match_font(names) is None)
+
+        # Check list of names.
+        names = [not_a_font, font, not_a_font]
+        font_path_2 = pygame_font.match_font(names)
+        self.assertEqual(font_path_2, font_path)
+        names = [not_a_font, not_a_font, not_a_font]
+        self.assertTrue(pygame_font.match_font(names) is None)
+
+        # Check generator:
+        names = (name for name in [not_a_font, font, not_a_font])
+        font_path_2 = pygame_font.match_font(names)
+        self.assertEqual(font_path_2, font_path)
+        names = (name for name in [not_a_font, not_a_font, not_a_font])
+        self.assertTrue(pygame_font.match_font(names) is None)
+
+        fonts_b = [f.encode() for f in font]
```

Comment:
Here, font variable is a string that contains the name of the font. By iterating over it and encoding it, you are encoding each character and putting it as a single element in the list. 
So fonts_b looks like 
`[b's', b'a', b'n', b's']`

Which is why CI is failing (I have no clue how pypy seems to pass)

Reply:
Thanks! Somehow missed that when testing on my machine.